core/lib/globals_api.h

```diff
@@ -156,6 +157,27 @@ typedef char bool;
 #    define INLINE_FORCED inline
 #    define WEAK __attribute__((weak))
 #    define NOINLINE __attribute__((noinline))
+/* As per https://gcc.gnu.org/onlinedocs/cpp/_005f_005fhas_005fattribute.html,
+ * we need to first check whether __has_attribute is defined, to ensure
+ * portability. We saw issues in the Android-ARM cross compile without this.
+ */
+#    if defined __has_attribute
+#        if __has_attribute(__no_sanitize__) && defined(USE_FNOSANITIZE_NULL)
```

Comment:
As the _api.h suffix denotes, this is a public header we give to clients.  These defines seem like something we only need internally.  I think it would be better to not put all this in this exposed header and instead keep it in globals.h or another internal header.

Reply:
> These defines seem like something we only need internally

Well, I thought `DISABLE_NULL_SANITIZER` could be useful to clients/extension libraries as well.

Reply:
Moved it to `globals.h`.